tensorflow_addons/optimizers/average_wrapper.py

```diff
@@ -16,11 +16,17 @@
 import abc
 
 import tensorflow as tf
+from typeguard import typechecked
 
 
 class AveragedOptimizerWrapper(tf.keras.optimizers.Optimizer, metaclass=abc.ABCMeta):
+    @typechecked
     def __init__(
-        self, optimizer, sequential_update=True, name="AverageOptimizer", **kwargs
+        self,
+        optimizer,
```

Comment:
```suggestion
        optimizer: tf.keras.optimizers.Optimizer,
```

Reply:
Thanks @gabrieldemarmiesse.

Reply:
Since Keras accepts optimizer class or string names of optimizers this will need to be a Union:
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/lookahead_test.py#L150

Reply:
Got it! Thanks @seanpmorgan.